tests/conftest.py

```diff
@@ -304,22 +252,72 @@ def df_type(request: pytest.FixtureRequest):
     yield df_type_str
 
 
+def _get_execution_mode(request: pytest.FixtureRequest) -> "ExecutionMode":
+    do_gpu_mode: bool = True
+
+    # Check for the param if this was indirectly set
+    if (hasattr(request, "param") and isinstance(request.param, bool)):
+        do_gpu_mode = request.param
+    else:
+        # If not, check for the marker and use that
+        gpu_mode = request.node.get_closest_marker("gpu_mode") is not None
+        cpu_mode = request.node.get_closest_marker("cpu_mode") is not None
+
+        if (gpu_mode and cpu_mode):
+            raise RuntimeError(f"Both markers (gpu_mode and cpu_mode) were added to function {request.node.nodeid}. "
+                               "Use the gpu_and_cpu_mode marker to test both.")
+
+        # if both are undefined, infer based on the df_type
+        if (not gpu_mode and not cpu_mode):
+            gpu_mode = request.node.get_closest_marker("use_cudf") is not None
```

Comment:
This line is never used. I think if we are going to infer the results, we should only infer if there is one or the other. Not both.

Reply:
It looks like you're right inferring gpu mode based on the `use_cudf` marker is not used. On the other hand, a large number of tests are relying upon the `use_pandas` marker for setting python/cpu mode.